crates/compiler/parse/src/expr.rs

```diff
@@ -2527,32 +2532,134 @@ fn list_literal_help<'a>() -> impl Parser<'a, Expr<'a>, EList<'a>> {
     .trace("list_literal")
 }
 
-pub fn record_value_field<'a>() -> impl Parser<'a, AssignedField<'a, Expr<'a>>, ERecord<'a>> {
-    use AssignedField::*;
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum RecordField<'a> {
+    RequiredValue(Loc<&'a str>, &'a [CommentOrNewline<'a>], &'a Loc<Expr<'a>>),
+    OptionalValue(Loc<&'a str>, &'a [CommentOrNewline<'a>], &'a Loc<Expr<'a>>),
+    LabelOnly(Loc<&'a str>),
+    SpaceBefore(&'a RecordField<'a>, &'a [CommentOrNewline<'a>]),
+    SpaceAfter(&'a RecordField<'a>, &'a [CommentOrNewline<'a>]),
+    ApplyValue(Loc<&'a str>, &'a [CommentOrNewline<'a>], &'a Loc<Expr<'a>>),
+}
```

Comment:
This is defined here because it's only used during parsing so that we don't have to backtrack. The AST defines a discrete `RecordBuilderField` type, which makes the desugaring process easier / faster.

The existing `AssignedField` type doesn't get this new `ApplyValue` variant.

Reply:
If you folks think there's a better way to write the parser without this intermediate type, please let me know. I did think of something but we'd need a custom parser instead of using `collection_inner!`.

Reply:
Personally +1 for including it, it adds effectively no runtime overhead and I think makes it clearer.